src/main/java/sg/edu/sutd/bank/webapp/service/ClientAccountDAOImpl.java

```diff
@@ -61,5 +65,57 @@ public void update(ClientAccount clientAccount) throws ServiceException {
 			closeDb(conn, ps, rs);
 		}
 	}
+	
+	/* get transaction details: amount, sender, receiver from client_transaction
+	   update sender and receiver alike
+	*/
+	public void executeTransaction(List<ClientTransaction> transactions) throws ServiceException {	
+		Connection conn = connectDB();
+		try {
+			for (ClientTransaction transaction : transactions) {
+				if (transaction.getStatus() == TransactionStatus.APPROVED) {
+					PreparedStatement ps = prepareStmt(conn, "SELECT * FROM client_transaction WHERE id = ?");
+					ps.setInt(1, transaction.getId()); // transaction-specific id
+					ResultSet rs = ps.executeQuery();
+					rs.next();
+					
+					// extracts USERID of sender & receiver
+					transaction.setFromAccount(rs.getInt("user_id"));
+					transaction.setToAccount(rs.getInt("to_account_num"));
+					BigDecimal amount = rs.getBigDecimal("amount");
+					
+					// Deduct money from sender's account
+					PreparedStatement psf = prepareStmt(conn, "UPDATE client_account SET amount = amount - ? WHERE user_id = ? AND amount >= ?");
+					psf.setBigDecimal(1, amount);
+					psf.setInt(2, transaction.getFromAccount());
+					psf.setBigDecimal(3, amount);
+					
+					// Insert money into recipient's account
```

Comment:
I think we should move lines 94-96 into the `if` block in line 105, because we only need to update the receiver amount if its valid.